src/server/game/Entities/Creature/TemporarySummon.cpp

```diff
@@ -35,7 +37,12 @@ m_timer(0), m_lifetime(0)
     m_unitTypeMask |= UNIT_MASK_SUMMON;
 }
 
-Unit* TempSummon::GetSummoner() const
+WorldObject* TempSummon::GetSummoner() const
+{
+    return m_summonerGUID ? ObjectAccessor::GetWorldObject(*this, m_summonerGUID) : nullptr;
+}
+
+Unit* TempSummon::GetSummonerUnit() const
 {
     return m_summonerGUID ? ObjectAccessor::GetUnit(*this, m_summonerGUID) : nullptr;
```

Comment:
I'd personally change this to
```cpp
if (WorldObject* summoner = GetSummoner())
    return summoner->ToUnit();
return nullptr;
```
because duplicating logic is never a good idea.

Reply:
Most of the changes were done in small steps to "migrate" the existing code to the new return types so I tried to change the least possible with each iteration.
It sounds good to do some codestyle changes too, maybe I can polish it up after it's confirmed that it works :)